compiler/optimizer/LoopReducer.cpp

```diff
@@ -2761,9 +2761,17 @@ TR_LoopReducer::generateArraytranslate(TR_RegionStructure * whileLoop, TR_Induct
             storeTree = storeTree->getNextTreeTop();
             }
          storeNode = (storeTree->getNode()->getNumChildren() == 2) ? storeTree->getNode() : NULL;
+
+         if (storeNode == NULL)
+            {
+            dumpOptDetails(comp(), "Store node is %p\n", storeNode);
+            return false;
+            }
+
          loadTree  = (storeNode != NULL && storeNode->getNumChildren() == 2) ? storeTree : NULL;
```

Comment:
wasn't the point of adding the if(storeNode == NULL) block above so that you wouldn't have to test if storeNode is NULL Here?